pkg/services/provisioning/provisioning.go

```diff
@@ -17,35 +17,41 @@ import (
 	"github.com/grafana/grafana/pkg/util/errutil"
 )
 
-type ProvisioningService interface {
-	ProvisionDatasources() error
-	ProvisionPlugins() error
-	ProvisionNotifications() error
-	ProvisionDashboards() error
-	GetDashboardProvisionerResolvedPath(name string) string
-	GetAllowUIUpdatesFromConfig(name string) bool
-}
-
 func init() {
 	registry.Register(&registry.Descriptor{
-		Name: "ProvisioningService",
-		Instance: newProvisioningServiceImpl(
-			dashboards.New,
-			notifiers.Provision,
-			datasources.Provision,
-			plugins.Provision,
-		),
+		Name:         "ProvisioningService",
+		Instance:     NewProvisioningServiceImpl(nil, nil, nil),
```

Comment:
Why do we need to pass `nil` references? 🤔 I'm not pretty sure which are the use cases we need to cover but we could make this function a bit more flexible if that's the case, I guess.

Reply:
Yes this relates to my comment just below. I don't like the way I did it :sweat: 
Either I initialize it in enterprise the same way it's done here in init (thus maybe one day, the initialization will defer)
Or I create a public constructor (which is what I did)
So what shall I do?
My only constraint is that I need to instantiate the service from outside the package.

Reply:
Anyway, I'd go for the easier solution without spending lot of time because I suspect that we're going to change this code sooner than later (due to the dependency injection refactoring). So, feel free 🚀  But, yeah, thanks a lot for clarifying it! 😊

Reply:
I'm going to do it the other way around to avoid this (nil,nil,nil) ugliness :sweat_smile: 
I'll have the NewProvisioningServiceImpl with no parameters initializing the private fields.
I'll populate the missing "injected" dependencies straight after the call to this constructor.